src/test/java/tests/FilterEvalTests.java

```diff
@@ -158,6 +156,22 @@ public void label() {
         } catch (ParseException ignored) {
         }
 
+        // Ensure that early termination does not occur
```

Comment:
Unfortunately, without the commit message, it took me a while to figure out what the "early termination" refers to (I thought it was early termination of the `label()` test at first, but that didn't make sense).

Is it possible to clarify that this refers to the new `labelsSatisfy` loop?